src/main/java/com/goncharenko/currencyexchangerate/service/impl/BankServiceImpl.java

```diff
@@ -61,23 +61,30 @@ public BankDTO retrieveById(Long id) {
             throw new ResourceNotFoundException("Bank with id " + id + " is not found");
         });
 
-        List<Currency> currencies = currencyRepository.retrieveAllCurrenciesByBankId(id).orElseThrow(() -> new ResourceNotFoundException("There are no currencies in this bank with id " + id));
+        List<Currency> currencies = currencyRepository.retrieveAllCurrenciesByBankId(id);
+        if (CollectionUtils.isEmpty(currencies)) {
+            LOGGER.debug("There are no currencies with bank id {} ", id);
+        }
         BankDTO bankDTO = BankDTO.convertToDTO(bank);
         bankDTO.setCurrencyDTOList(currencies.stream().map(CurrencyDTO::convertToDTO).collect(Collectors.toList()));
-        LOGGER.info("Bank with id {} was retrieved" + id);
+        LOGGER.info("Bank with id {} was retrieved", id);
         return bankDTO;
     }
 
     @Transactional
     @Override
     public List<BankDTO> retrieveAll() {
-        List<Bank> retrievedBanks = bankRepository.retrieveAll().orElseThrow(() -> {
+        List<Bank> retrievedBanks = bankRepository.retrieveAll();
+        if (CollectionUtils.isEmpty(retrievedBanks)) {
             LOGGER.debug("There are no banks in table ");
             throw new ResourceNotFoundException("There are no banks in table");
-        });
+        }
```

Comment:
I did not check it in "service" task  if service method that returns a list and list is empty it is ok to return Empty List. Change the behavior.